main.rb

```diff
@@ -0,0 +1,175 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    y = 0
+    arr = to_a
+    while y < arr.length
+      yield arr[y]
+      y += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    y = 0
+    arr = to_a
+    while y < arr.length
+      yield arr[y], y
+      y += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    access = []
+    my_each do |val|
+      access << val if yield val
+    end
+    access
+  end
+
+  def my_all?(par = nil)
+    return test_parm(par) unless par.nil?
+
+    return (my_all? { |x| !x.nil? && x != false }) unless block_given?
+
+    my_each do |a|
+      return false unless yield a
+    end
+    true
+  end
+
+  def test_parm(par)
+    if par.class == Class
+      my_all? { |x| x.class == par }
+    elsif par.class == Regexp
+      my_all? { |x| x =~ par }
+    else
+      my_all? { |x| x == par }
+    end
+  end
+
+  def my_any?(par = nil)
+    return test_param_any(par) unless par.nil?
+
+    return (my_any? { |x| !x.nil? && x != false }) unless block_given?
+
+    my_each do |a|
+      return true if yield a
+    end
+    false
+  end
+
+  def test_param_any(par)
+    if par.class == Class
+      my_any? { |x| x.class == par }
+    elsif par.class == Regexp
+      my_any? { |x| x =~ par }
+    else
+      my_any? { |x| x == par }
+    end
+  end
+
+  def my_none?(par = nil)
+    return test_par_none(par) unless par.nil?
+
+    return (my_none? { |x| !x.nil? && x != false }) unless block_given?
+
+    my_each do |a|
+      return false if yield a
+    end
+    true
+  end
```

Comment:
[OPTIONAL]

I think it's possible to get my_none? from my_any? we should only do something like !my_any? for my_none? to work For example:
```
  def my_none?(arg = nil, &block)
    !my_any?(arg = nil,&block)
  end
```
To make this work we have to use block.call(element) instead of yield(element) in my_any?